tests/python/cli/test_cli.py

```diff
@@ -243,23 +242,26 @@ def mocked_version(_):
         assert "Server version '0' is not compatible with SDK version" in caplog.text
 
     @pytest.mark.parametrize("verify", [True, False])
-    def test_can_control_ssl_verification_with_arg(self, monkeypatch, verify: bool):
-        # TODO: Very hacky implementation, improve it, if possible
-        class MyException(Exception):
-            pass
-
-        normal_init = CLI.__init__
-
-        def my_init(self, *args, **kwargs):
-            normal_init(self, *args, **kwargs)
-            raise MyException(self.client.api_client.configuration.verify_ssl)
-
-        monkeypatch.setattr(CLI, "__init__", my_init)
-
-        with pytest.raises(MyException) as capture:
-            self.run_cli(*(["--insecure"] if not verify else []), "ls")
-
-        assert capture.value.args[0] == verify
+    def test_can_control_ssl_verification_with_arg(self, verify: bool):
+        with https_reverse_proxy() as proxy_url:
+            if verify:
+                insecure_args = []
+            else:
+                insecure_args = ["--insecure"]
+
+            run_cli(
+                self,
+                f"--auth={self.user}:{self.password}",
+                f"--server-host={proxy_url}",
+                *insecure_args,
+                "ls",
+                expected_code=1 if verify else 0,
+            )
+            stdout = self.stdout.getvalue()
+
+        if not verify:
+            for line in stdout.splitlines():
+                int(line)
```

Comment:
```suggestion
        if not verify:
            assert list(map(int, stdout.splitlines()))
        else:
            assert "certificate verify failed" in stdout
```

Reply:
I don't want to depend on a 3rd-party library's error messages; that's not part of the API and could change with no notice.

IMO, testing that the call succeeds with `--insecure`, but not without it gives sufficient confidence that the failure is due to a certificate problem.